src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,142 @@
-
-    skeleton = skeletize(thresh4, size, skel)
-
-    dst = cv2.Laplacian(skeleton, ddepth, ksize=kernel_size)
-    abs_dst = cv2.convertScaleAbs(dst)
-
-    linesP = cv2.HoughLinesP(abs_dst, rho=6, theta=np.pi / 60, threshold=50,
-                             lines=np.array([]), minLineLength=30, maxLineGap=10)
-
-    # x_array_left = [[]]
-    # x_array_right = [[]]
-    # print(x_array.shape)
-    # y_array_left = [[]]
-    # y_array_right = [[]]
-    # print(y_array.shape)
-
-    crop2 = crop
-
-    if linesP is not None and len(linesP) > 40:
-        # processed = process_lines(linesP,crop)
-        # print(len(linesP))
-        for line in linesP:
+        for line in lines:
             for x1, y1, x2, y2 in line:
-                slope = (y2 - y1) / (x2 - x1)
-                # <-- Calculating the slope.
-                if math.fabs(slope) < .5:
-                    # <-- Only consider extreme slope
-                    continue
-                if x1 < frame.shape[1]/2 and x2 < frame.shape[1]/2:
-                    # <-- If the slope is negative, left group
-                    # x_array_left.append([x1,x2])
-                    # y_array_left.append([y1,y2])
-                    cv2.line(crop2, (x1, y1), (x2, y2), (0, 0, 255), 3)
-                    # cv2.circle(crop, (x1, y1),(5), (0,0,255), 3)
-                    # cv2.circle(crop, (x2, y2),(5), (0,0,255), 3)
-                else:  # <-- Otherwise, right group.
-                    # x_array_right.append([x1,x2])
-                    # y_array_right.append([y1,y2])
-                    cv2.line(crop2, (x1, y1), (x2, y2), (0, 255, 0), 3)
-                    # cv2.circle(crop, (x1, y1),(5), (0,255,0), 3)
-                    # cv2.circle(crop, (x2, y2),(5), (0,255,0), 3)
-            circles = crop
-            lines = crop2
-
-    if linesP is None or len(linesP) < 40:
-        skipped_frames += 1
-
-    cv2.imshow('Crop', crop)
-    # linesAndCircles = cv2.addWeighted(lines, 0.5, circles, 0.5,0)
-    # final = cv2.addWeighted(processed, 0.8, linesAndCircles, 0.5,0)
-    # cv2.imshow('Original', frame)
-    # cv2.imshow('Combo', final)
-    # print(skipped_frames)
-
-    # FIXME: what is this crap, please FIX ME
-    # plt.plot(np.unique(x_array_right),
-    #          np.poly1d(np.polyfit(x_array_right, y_array_right, 1))
-    #          (np.unique(x_array_right)))
-
-    # right_line_slope,right_line_intercept = best_fit(np.asarray(x_array_right),np.asarray(y_array_right))
-    # left_line_slope,left_line_intercept = best_fit(np.asarray(x_array_left),np.asarray(y_array_left))
-
-    toc = time.time()
-    delta_t = toc - tic
-    print("FPS: {}".format(1/delta_t))
-
-    key = cv2.waitKey(25)
-    if key == 27:
-        break
-
-video.release()
-cv2.destroyAllWindows
+                cv2.line(line_img, (x1, y1), (x2, y2), color, thickness)
+        img = cv2.addWeighted(img, 0.8, line_img, 1.0, 0.0)
+        return img
+
+    def best_fit(x_points, y_points):
```

Comment:
Same as above for private variables and methods. Add a `_` at the beginning. Ex. `self._type_of_obstacle`